### PR TITLE
SNOW-507942 handle s3 accelerate configuration correctly

### PR DESCRIPTION
**Pull request by sfc-gh-cshi**
A test is added to make sure right endpoint is used according to accelerate configuration

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-507942

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

  1)  Check s3 accelerate config to use right URL; 2) parse XML response to retrieve configuration status correctly; 3) make sure the accelerate config could be updated correctly from previous s3 API call to all s3 client objects where they have their own endpoint.
  
  I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

**Comment by sfc-gh-cshi** [comment]
I have read the CLA Document and I hereby sign the CLA

File: test/integ/test_put_get_user_stage.py [hunk 1]
```diff
@@ -39,6 +40,72 @@ def test_put_get_small_data_via_user_stage(is_public_test, tmpdir, conn_cnx, fro
...
+    "accelerate_config",
+    [True, False],
+)
+def test_put_get_accelerate_user_stage(
+    is_public_test, tmpdir, conn_cnx, from_path, accelerate_config
+):
+    from snowflake.connector.file_transfer_agent import SnowflakeFileTransferAgent
+    from snowflake.connector.s3_storage_client import SnowflakeS3RestClient
+
+    """[s3] Puts and Gets Small Data via User Stage."""
```
Comment (sfc-gh-mkeller) [inline test/integ/test_put_get_user_stage.py:59]: This has to be the first thing in the function to be considered a doc-string, right now it's just a string constant that nothing points to

File: test/integ/test_put_get_user_stage.py [hunk 2]
```diff
@@ -39,6 +40,72 @@ def test_put_get_small_data_via_user_stage(is_public_test, tmpdir, conn_cnx, fro
...
+    if is_public_test or "AWS_ACCESS_KEY_ID" not in os.environ:
```
Comment (sfc-gh-mkeller) [inline test/integ/test_put_get_user_stage.py:60]: I think you can just [tag the test as an internal test](https://github.com/snowflakedb/snowflake-connector-python/blob/f2f7e6f35a162484328399c8a50a5015825a5573/test/integ/test_cursor_binding.py#L139) to achieve the same thing

File: test/unit/test_s3_util.py [hunk 3]
```diff
@@ -106,13 +106,20 @@ def test_upload_file_with_s3_upload_failed_error(tmp_path):
         },
     )
     exc = Exception("Stop executing")
+
+    def mock_transfer_accelerate_config(
+        self: SnowflakeS3RestClient, use_accelerate_endpoint: bool = None
```
Comment (sfc-gh-mkeller) [inline test/unit/test_s3_util.py:111]:
```suggestion
        self: SnowflakeS3RestClient,
        use_accelerate_endpoint: bool | None = None,
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]

**Pull request merged**